src/Processors/Formats/Impl/ValuesBlockInputFormat.h

```diff
@@ -37,7 +37,18 @@ class ValuesBlockInputFormat final : public IInputFormat
     void resetReadBuffer() override;
 
     /// TODO: remove context somehow.
-    void setContext(const ContextPtr & context_) { context = Context::createCopy(context_); }
+    void setContext(const ContextPtr & context_)
+    {
+        auto context_copy = Context::createCopy(context_);
+
+        /// ConstantExpressionTemplate generates placeholder names (_dummy_N)
+        /// for all literals, which are valid names for creating named tuples.
+        /// This behavior needs to be explicitly disabled, because if named
+        /// tuples with different names are inserted into a named tuple, it will
+        /// only insert default values.
```

Comment:
Do you mean something like
```
CREATE TABLE tbl(x Tuple(a Int32, b Int32, c Int32)) ENGINE=MergeTree ORDER BY tuple();
INSERT INTO tbl VALUES ((1,2,3));
```
?

And without this change it would end up with inserting just `(0, 0, 0)` instead of `(1, 2, 3)`?

Reply:
And what about

```
CREATE TABLE tbl(x Tuple(a Int32, b Int32, c Int32)) ENGINE=MergeTree ORDER BY tuple();
INSERT INTO tbl VALUES ((1 AS b,2 AS c,3 AS a));
```
?

Reply:
The following will end up with `(0, 0, 0)`
```
CREATE TABLE tbl(x Tuple(a Int32, b Int32, c Int32)) ENGINE=MergeTree ORDER BY tuple();
INSERT INTO tbl VALUES (tuple(1,2,3));   -- without tuple it's interpreted differently inside values block.
```

`INSERT INTO tbl VALUES (tuple(1 AS b,2 AS c,3 AS a));`   also `(0, 0, 0)`.
`INSERT INTO tbl VALUES (tuple(1 AS a, 2 AS b,3 AS c));`   also `(0, 0, 0)`. Alias in values block doesn't work at all.

Reply:
We do need a test about this case.

Reply:
And please add your first example
```
CREATE TABLE tbl(x Tuple(a Int32, b Int32, c Int32)) ENGINE=MergeTree ORDER BY tuple();
INSERT INTO tbl VALUES (tuple(1,2,3));   -- without tuple it's interpreted differently inside values block.
```
to the comment in the code.

Reply:
Sure. And l will use a different approach to work around this issue without disabling the setting. Instead of using `_dummy_` as the prefix for those columns, we can use `-dummy-`. This way, the identifiers won't be treated as unquoted identifiers and are theoretically less likely to conflict with human input names. We've used similar synthetic names in old join rewrites.